python/rules_engine.py

```diff
@@ -2644,6 +2643,136 @@ def v708(self):
+				(self.lar_df.app_score_code_8!="")&(self.lar_df.co_app_score_name!="")&(self.lar_df.co_app_score_code_8!=""))]
+		self.results_wrapper(edit_name=edit_name, field_name=field, fail_df=fail_df)
+
+	def v711_1(self):
+		"""1) If the Reason for Denial exemption election is
+			taken, Reason for Denial: 1 must be reported 1111;"""
+		field = "Reason for Denial"
+		edit_name = "v711_1"
+		fail_df = self.lar_df[(self.lar_df.denial_1=="1111")&(self.lar_df.denial_1!="1111")]
+		self.results_wrapper(edit_name=edit_name, field_name=field, fail_df=fail_df)
+
+	def v711_2(self):
+		"""2)If the Reason for Denial exemption election is
+			taken, 
+			Reason for Denial: 2, Reason for Denial: 3,
+			Reason for Denial: 4, and Reason for Denial:
+			Conditional Free Form Text Field for Code 9 must be
+			left blank."""
+		field = "Reason for Denial"
+		edit_name = "v711_2"
+		fail_df = self.lar_df[(self.lar_df.denial_1=="1111")&((self.lar_df.denial_2!="")&(self.lar_df.denial_3!="")
+					&(self.lar_df.denial_4!="")&(self.lar_df.denial_code_9!=""))]
+		self.results_wrapper(edit_name=edit_name, field_name=field, fail_df=fail_df)
+
+	def v712(self):
+		"""1) If the Total Loan Costs or Total Points and Fees
+				exemption election is taken, Total Loan Costs and
+				Total Points and Fees must be reported Exempt."""
+		field = "Total Loan Costs/Points and Fees"
+		edit_name = "v712"
+		fail_df = self.lar_df[((self.lar_df.loan_costs=="Exempt") | (self.lar_df.points_fees=="Exempt")) & ((self.lar_df.loan_costs!="Exempt")
+					| (self.lar_df.points_fees!="Exempt"))]
+		self.results_wrapper(edit_name=edit_name, field_name=field, fail_df=fail_df)
+
+	def v713_1(self):
+		"""If the Automated Underwriting System exemption
+			election is taken,
+			1) Automated Underwriting System: 1 and
+			Automated Underwriting System Result: 1 must be
+			reported 1111; and"""
+		field = "Automated Underwriting System"
+		edit_name = "v713_1"
+		fail_df = self.lar_df[(self.lar_df.aus_1=="1111") & ((self.lar_df.aus_1!="1111") & (self.lar_df.aus_result_1!="1111"))]
+		self.results_wrapper(edit_name=edit_name, field_name=field, fail_df=fail_df)
+	
+	def v713_2(self):
+		"""If the Automated Underwriting System exemption
+			election is taken,
+			2) Automated Underwriting System: 2, Automated
+			Underwriting System: 3, Automated Underwriting
+			System: 4, Automated Underwriting System: 5,
+			Automated Underwriting System: Conditional Free
+			Form Text Field for Code 5, Automated Underwriting
+			System Result: 2, Automated Underwriting System
+			Result: 3, Automated Underwriting System Result: 4,
+			Automated Underwriting System Result: 5, and
+			Automated Underwriting System Result: Conditional
+			Free Form Text Field for Code 16 must be left blank."""
+		field = "Automated Underwriting System"
+		edit_name = "v713_2"
+		fail_df = self.lar_df[(self.lar_df.aus_1=="1111") & ((self.lar_df.aus_2!="") & (self.lar_df.aus_3!="") & (self.lar_df.aus_4!="") & 
```

Comment:
I think using ors in the second clause will give better results as it will check each clause independently. Since taking the exemption requires all to be blank, any one of the other fields not being blank is enough to cause a fail.